docs: flow dev guide (#959)
GreptimeTeam/docs: docs/nightly/en/contributor-guide/flownode/overview.md

# Overview

## Introduction

> `Flownode` provide simple streaming process(known as `flow`) ability to the database. 

Reviewer: ```suggestion
`Flownode` provides a simple streaming process (known as `flow`) ability to the database. 
```

> `Flownode` manage `flows` which are tasks that actively receive data from the `source` and send data to the `sink`.

Reviewer: ```suggestion
`Flownode` manages `flows` which are tasks that actively receive data from the `source` and send data to the `sink`.
```

> In current version, `Flownode` only support standalone mode. In the future, we will support distributed mode.

Reviewer: ```suggestion
In the current version, `Flownode` only supports standalone mode. In the future, we will support distributed mode.
```

## Components

> A `Flownode` contains all the components needed for flow's streaming process, here we list vital parts:

Reviewer: ```suggestion
A `Flownode` contains all the components needed for the streaming process of a flow. Here we list the vital parts:
```

> - A `FlownodeManager` for receiving inserts being forward from `Frontend` and send back results for flow's sink table

Reviewer: ```suggestion
- A `FlownodeManager` for receiving inserts forwarded from the `Frontend` and sending back results for the flow's sink table.
```

> - Certain number of `FlowWorker` each running in a separate thread, currently for standalone the number of flow worker is one, but prone to change in the future

Reviewer: ```suggestion
- A certain number of `FlowWorker` instances, each running in a separate thread. Currently for standalone mode, there is only one flow worker, but this may change in the future.
```

> - A `Flow` is a task that actively receive data from the `source` and send data to the `sink`, it is managed by `FlownodeManager` and run by `FlowWorker`.

Reviewer: ```suggestion
- A `Flow` is a task that actively receives data from the `source` and sends data to the `sink`. It is managed by the `FlownodeManager` and run by a `FlowWorker`.
```
